src/createNavigationContainer.js

```diff
@@ -152,6 +152,13 @@ export default function createNavigationContainer<T: *>(
       this._validateProps(nextProps);
     }
 
+    componentDidUpdate() {
+      // Clear cached _nav every tick
+      if (this._nav === this.state.nav) {
+        delete this._nav;
```

Comment:
I can make the change. But this is going to be a pointless level of micro-optimization on a class like NavigationContainer; and given the React+Babel weight I have strong doubts that this class' instances are "hot" objects that use that one type of optimization.